geniza/corpus/forms.py

```diff
@@ -334,6 +336,61 @@ def clean(self):
             self.add_error(
                 "q", _("Relevance sort is not available without a keyword search term.")
             )
+        # additional validation for regex mode due to some queries that cause Lucene errors
+        mode = cleaned_data.get("mode")
+        if mode == "regex":
```

Comment:
oh wow, this is really detailed error checking and help messages! would be interesting to know how much these come up, but I'm sure the people who trip over them will find it useful